common/src/main/scala/au/csiro/data61/magda/model/Registry.scala

```diff
@@ -0,0 +1,81 @@
+package au.csiro.data61.magda.model
+
+import io.swagger.annotations.{ ApiModel, ApiModelProperty }
+import enumeratum.values.{ IntEnum, IntEnumEntry }
+import scala.annotation.meta.field
+import spray.json.JsObject
+import java.time.OffsetDateTime
+import spray.json.DefaultJsonProtocol
+import spray.json.JsValue
+import spray.json.JsNumber
+import spray.json.RootJsonFormat
+
+object Registry {
+  @ApiModel(description = "A type of aspect in the registry.")
+  case class AspectDefinition(
+    @(ApiModelProperty @field)(value = "The unique identifier for the aspect type.", required = true) id: String,
+
+    @(ApiModelProperty @field)(value = "The name of the aspect.", required = true) name: String,
+
+    @(ApiModelProperty @field)(value = "The JSON Schema of this aspect.", required = false, dataType = "object") jsonSchema: Option[JsObject])
+
+  case class WebHookPayload(
```

Comment:
As you may have gathered, I'm kind of fond of the one-file-per-type thing.  But is that crazy cause of Scala compilation times or something?